sound/soc/sof/sof-client.h

```diff
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+ */
+
+#ifndef __SOUND_SOC_SOF_CLIENT_H
+#define __SOUND_SOC_SOF_CLIENT_H
+
+#include <linux/virtual_bus.h>
+
+enum sof_client_type {
+	SOF_CLIENT_AUDIO,
+	SOF_CLIENT_IPC,
+};
+
+/* SOF client device */
+struct sof_client_dev {
+	struct virtbus_device vdev;
+	void *data;
+};
+
+struct sof_client_drv {
+	const char *name;
+	enum sof_client_type type;
+	struct sof_client_ops *ops;
+	struct virtbus_driver *virtbus_drv;
```

Comment:
don't you want a 'struct virtbus_driver virtbus_drv' here?
otherwise you can't use container_of, and the encapsulation is hard to deal with.

Reply:
@plbossart that is true but I can imagine the need for dealing with container_of for the client drv at all. the sof_client_dev has a virtbus_device member. I can only imagine that we'd need to use the container_of to get the client_dev from the virtbus_device but not the driver itself.

Reply:
nevertheless I've changed it now.